File: coding-test/nodejs/src/practice/fibonacci-number.js
// @ts-check
function solution(n) {
  let answer = 0
  let n_2 = 0
  let n_1 = 1
  for (let i = 0; i < n - 1; i += 1) {
    answer = (n_1 + n_2) % 1234567
    n_2 = n_1
    n_1 = answer
  }

  return answer
}

const n = 5
console.log(solution(n))

File: coding-test/java-kotlin/src/kr/co/programmers/java/practice/FibonacciNumber.java
package kr.co.programmers.java.practice;

public class FibonacciNumber {

    public static void main(String[] args) {
        int n = 3;
        System.out.println(new Solution().solution(n));
    }

    static class Solution {
        public int solution(int n) {
            int answer = 0;
            int n_2 = 0;
            int n_1 = 1;
            for(int i = 0; i < n-1; i++){
                answer = (n_1 + n_2) % 1234567;
                n_2 = n_1;
                n_1 = answer;
            }

            return answer;
        }
    }
}
